apps/api/src/goods-receipts/goods-receipts.service.ts

```diff
@@ -0,0 +1,75 @@
+import { Injectable } from '@nestjs/common';
+import { CreateGoodsReceiptDto } from './dto/create-goods-receipt.dto';
+import { UpdateGoodsReceiptDto } from './dto/update-goods-receipt.dto';
+import { PrismaService } from "../prisma.service";
+import { GoodsReceipt } from '@prisma/client';
+
+@Injectable()
+export class GoodsReceiptsService {
+  constructor(private readonly prisma: PrismaService) {}
+
+  async create(createGoodsReceiptDto: CreateGoodsReceiptDto) {
+    return this.prisma.goodsReceipt.create({
+      data: {
+        purchase_order_id: createGoodsReceiptDto.purchase_order_id,
+        received_at: createGoodsReceiptDto.received_at,
+        line_items: {
+          create: createGoodsReceiptDto.line_items.map(item => ({
+            item_id: item.item_id,
+            quantity_received: item.quantity_received
+          }))
+        },
+      },
+    });
```

Comment:
No right answer here but do you have any typical code boundaries around this layer and the data? For example, do you like or ever use a data access object to abstract away or centralize the data operations and have them be different that whats in the service layer or is that overkill and so talking directly to the ORM here makes more sense?

Reply:
I do think it is overkill for simple applications (like this exercise) and can cause unnecessary overhead for developers by adding an additional layer. 

However, if you foresee the application being complex or the need to share data access across services, it definitely helps with reusability and maintainability. 

I have been part of both approaches in past projects. So, in summary it depends 🤷‍♂️

Reply:
Are you using DAOs at GoodDay?